### PR TITLE
core: Proper vehicle fix

### PR DESCRIPTION
**Pull request by ls-**
@Nevcairiel <3

### CODE REVIEW

File: oUF.xml [hunk 1]
```diff
@@ -39,10 +39,12 @@
...
-	<Button name="oUF_ClickCastUnitTemplate" virtual="true" inherits="SecureUnitButtonTemplate, SecureHandlerEnterLeaveTemplate">
+	<Button name="oUF_ClickCastUnitTemplate" virtual="true" inherits="SecureUnitButtonTemplate, SecureHandlerStateTemplate, SecureHandlerEnterLeaveTemplate">
 		<Attributes>
 			<Attribute name="_onenter" type="string" value="local snippet = self:GetAttribute('clickcast_onenter'); if snippet then self:Run(snippet) end"/>
 			<Attribute name="_onleave" type="string" value="local snippet = self:GetAttribute('clickcast_onleave'); if snippet then self:Run(snippet) end"/>
+			<Attribute name="refreshUnitChange" type="string" value="local unit = self:GetAttribute('unit'); if unit then RegisterStateDriver(self, 'vehicleui', ('[@%s,unithasvehicleui]vehicle; novehicle'):format(unit)) else UnregisterStateDriver(self, 'vehicleui') end"/>
+			<Attribute name="_onstate-vehicleui" type="string" value="local unit = self:GetAttribute('unit'); if unit and newstate == 'vehicle' and UnitPlayerOrPetInRaid(unit) and not UnitTargetsVehicleInRaidUI(unit) then self:SetAttribute('toggleForVehicle', false) else self:SetAttribute('toggleForVehicle', true) end"/>
```
Comment (p3lim) [inline oUF.xml]: `self:SetAttribute('toggleForVehicle', not(unit and newstate == 'vehicle' and UnitPlayerOrPetInRaid(unit) and not UnitTargetsVehicleInRaidUI(unit)))`

Comment (haste) [reply oUF.xml]: Could you make it any more unreadable? :P

Comment (p3lim) [reply oUF.xml]: Was better than the current one, no if-else and repetition of the SetAttribute call.
It's ugly regardless.

Comment (ls-) [reply oUF.xml]: It's ugly regardless :p:gun:

**Pull request merged**